rtl/eclrtl/eclrtl.cpp

```diff
@@ -2318,6 +2360,39 @@ void rtlTrimAll(unsigned & tlen, char * & tgt, unsigned slen, const char * src)
     tgt = buffer;
 }
 
+void rtlTrimWS(unsigned & tlen, char * & tgt, unsigned slen, const char * src, bool left, bool all, bool right)
+{
+    if (all)
+    {
+        tlen = rtlTrimStrLenNonWhitespace(slen, src);
+        char * buffer = (char *)rtlMalloc(tlen + 1);
+        int ind = 0;
+        for(unsigned i = 0; i < slen; i++) {
+            if (!isspace(src[i]))
+            {
+                buffer[ind] = src[i];
+                ind++;
+            }
+        }
+        buffer[tlen] = 0;
```

Comment:
There shouldn't be any need to zero terminate the result.

Reply:
Probably the person who originally coded the trim variants (not sure who) didn't realise this.